### PR TITLE
add tabs in zh-cn/overview/scala3-book/num9

### PR DESCRIPTION
**Pull request by benluo**

### CODE REVIEW

File: _zh-cn/overviews/scala3-book/taste-modeling.md [hunk 1]
```diff
@@ -32,14 +32,38 @@ It’s more about modeling operations, imho.”
...
 稍后，当您想要创建属性和行为的具体实现时，类和对象可以扩展特征，根据需要混合尽可能多的特征以实现所需的行为。
 
 作为如何将 traits 用作接口的示例，以下是三个 traits，它们为狗和猫等动物定义了结构良好并且模块化的行为：
 
+{% tabs traits class=tabs-scala-version %}
+{% tab 'Scala 2' for=traits %}
+
+```scala
+trait Speaker {
+  def speak(): String  // has no body, so it’s abstract
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/taste-modeling.md:48]: FYI I noticed that in the scala 3 tab this comment was translated if you want to copy that over

File: _zh-cn/overviews/scala3-book/taste-modeling.md [hunk 2]
```diff
@@ -120,20 +237,57 @@ to replace the Scala2 “sealed trait + case class” pattern. How to resolve?
...
 - 面饼类型
 - 馅料
 
-这些是用枚举简洁地建模的：
+这些是用枚举简洁地建模的，它们是只包含单例值的 sum types：
+
+{% tabs enum_1 class=tabs-scala-version %}
+{% tab 'Scala 2' for=enum_1 %}
+
+In Scala 2 `sealed` classes and `case object` are combined to define an enumeration:
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/taste-modeling.md:260]: not translated, but I'm ok to accept

**Review by bishabosha — APPROVED** [review]
The code tabs work correctly, there's still some English left untranslated if that was unintended

File: _zh-cn/overviews/scala3-book/taste-modeling.md (hunk 1 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/taste-modeling.md:48]: change to untranslated version in scala3 tab. I prefer keep code without translation.

File: _zh-cn/overviews/scala3-book/taste-modeling.md (hunk 2 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/taste-modeling.md:260]: translated.

File: _zh-cn/overviews/scala3-book/taste-modeling.md [hunk 3]
```diff
@@ -120,20 +237,57 @@ to replace the Scala2 “sealed trait + case class” pattern. How to resolve?
...
 - 面饼类型
 - 馅料
 
-这些是用枚举简洁地建模的：
+这些是用枚举简洁地建模的，它们是只包含单例值的 sum types：
+
+{% tabs enum_1 class=tabs-scala-version %}
+{% tab 'Scala 2' for=enum_1 %}
+
+在 Scala 2 中，用 `sealed` 类和 `case object` 组合在一起来定义枚举： are combined to define an enumeration:
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/taste-modeling.md:260]:
```suggestion
在 Scala 2 中，用 `sealed` 类和 `case object` 组合在一起来定义枚举:
```

**Pull request merged**